core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -1389,7 +1389,8 @@ struct ViewOffset<
     KOKKOS_INLINE_FUNCTION
     static constexpr size_t stride(size_t const N) {
       return ((align != 0) &&
-              ((Kokkos::Impl::MEMORY_ALIGNMENT_THRESHOLD * align) < N) &&
+              ((static_cast<int>(Kokkos::Impl::MEMORY_ALIGNMENT_THRESHOLD) *
+                static_cast<int>(align)) < N) &&
```

Comment:
how does this not warn? Its a comparison of an int with a size_t? Shouldn't this case to size_t?